src/parse.rs

```diff
@@ -15,8 +15,8 @@ pub trait Parse<Cfg> {
     fn parse(&mut Parser, Cfg) -> Self;
 }
 
-impl Parse<bool> for Test {
-    fn parse(parser: &mut Parser, failing: bool) -> Test {
+impl Parse<(bool, bool)> for Test {
```

Comment:
Can you instead wrap these two bools in a simple struct `TestConfig { ignored: bool, failing: bool }` and pass that around instead? I think that will make the code a little clearer and less prone to subtle bugs in the future.